src/phpFastCache/Drivers/Zendshm/Driver.php

```diff
@@ -0,0 +1,140 @@
+ * Requires Zend Data Cache Functions from ZendServer
+ * @package phpFastCache\Drivers
+ */
+class Driver extends DriverAbstract
+{
+    /**
+     * Driver constructor.
+     * @param array $config
+     * @throws phpFastCacheDriverException
+     */
+    public function __construct(array $config = [])
+    {
+        $this->setup($config);
+
+        if (!$this->driverCheck()) {
+            throw new phpFastCacheDriverCheckException(sprintf(self::DRIVER_CHECK_FAILURE, $this->getDriverName()));
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    public function driverCheck()
+    {
+        if (extension_loaded('Zend Data Cache') && function_exists('zend_shm_cache_store')) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @param \Psr\Cache\CacheItemInterface $item
+     * @return mixed
+     * @throws \InvalidArgumentException
+     */
+    protected function driverWrite(CacheItemInterface $item)
+    {
+        /**
+         * Check for Cross-Driver type confusion
+         */
+        if ($item instanceof Item) {
+            $ttl = $item->getExpirationDate()->getTimestamp() - time();
+
+            return zend_shm_cache_store($item->getKey(), $this->driverPreWrap($item), ($ttl > 0 ? $ttl : 0));
+        } else {
+            throw new \InvalidArgumentException('Cross-Driver type confusion detected');
+        }
+    }
+
+    /**
+     * @param \Psr\Cache\CacheItemInterface $item
+     * @return mixed
+     */
+    protected function driverRead(CacheItemInterface $item)
+    {
+        $data = zend_shm_cache_fetch($item->getKey());
+        if ($data === false) {
+            return null;
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param \Psr\Cache\CacheItemInterface $item
+     * @return bool
+     * @throws \InvalidArgumentException
+     */
+    protected function driverDelete(CacheItemInterface $item)
+    {
+        /**
+         * Check for Cross-Driver type confusion
+         */
+        if ($item instanceof Item) {
+            return zend_shm_cache_delete($item->getKey());
+        } else {
+            throw new \InvalidArgumentException('Cross-Driver type confusion detected');
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    protected function driverClear()
+    {
+        return @zend_shm_cache_clear();
+    }
+
+    /**
+     * @return bool
+     */
+    protected function driverConnect()
+    {
+        return true;
+    }
+
+    /********************
+     *
+     * PSR-6 Extended Methods
+     *
+     *******************/
+
+    /**
+     * @return driverStatistic
+     */
+    public function getStats()
+    {
+        $stats = (array) zend_shm_cache_info();
+        return (new driverStatistic())
+            ->setData(implode(', ', array_keys($this->namespaces)))
+            ->setInfo(sprintf("The Zend memory have %d item(s) in cache.\n For more information see RawData.",$stats[ 'items_total' ]))
```

Comment:
Like Scrutinizer [said](https://scrutinizer-ci.com/g/PHPSocialNetwork/phpfastcache/inspections/eb8f658f-5772-4522-9449-fcccd7dc4eed/issues/files/src/phpFastCache/Drivers/Zendshm/Driver.php?status=new&orderField=path&order=asc&honorSelectedPaths=0):

I think that you made a mistake here: The property `$this->namespaces` does not exists, you may want to use `$this->itemInstances` instead ?